library/Booster/Pattern/Base.hs

```diff
@@ -723,7 +723,7 @@ makeBaseFunctor ''Predicate
 -- | A term (configuration) constrained by a number of predicates.
 data Pattern = Pattern
     { term :: Term
-    , constraints :: ![Predicate]
+    , constraints :: Set Predicate
```

Comment:
!Haskell noob question alert!
Why are we dropping the strictness annotation here?

Reply:
The `Set` type is strict by itself.
The bang `!` just puts demand on the entire data structure so it might actually make sense to keep it.
I will just put it back, just in case.